src/scripts/main.js

```diff
@@ -1,3 +1,231 @@
+  const userSalary = newData.get('salary');
+
+  if (userName.length < 4) {
+    pushNotification('error', 'Error', 'Name must have more than 3 letters');
+  } else if (userAge < 18 || userAge > 90) {
+    pushNotification('warning', 'Warning', 'Age is incorrect !');
+  } else {
+    tbody.insertAdjacentHTML('afterbegin', `
+      <tr>
+        <td>${newData.get('name')}</td>
+        <td>${newData.get('position')}</td>
+        <td>${newData.get('office')}</td>
+        <td>${newData.get('age')}</td>
+        <td>${newSalary(userSalary)}</td>
+      </tr>
+    `);
+    pushNotification('success', 'Success', 'A new employee has been added !');
+    form.reset();
+  }
+});
+
+tbody.addEventListener('click', (ev) => {
+  const selected = ev.target.closest('tr');
+
+  [...tbody.children].forEach((row) => {
+    row.classList.remove('active');
+  });
+  selected.classList.add('active');
+});
+
+const headTitle = document.querySelector('thead');
+let prevIndex = 0;
+
+headTitle.addEventListener('click', ev => {
+  const titleIndex = ev.target.cellIndex;
+  const sorted = [...tbody.children].sort((a, b) => {
+    let sortA = a.cells[titleIndex].innerText;
+    let sortB = b.cells[titleIndex].innerText;
+
+    if (sortA.includes('$')) {
+      sortA = sortA.replace(/[$,]/g, '');
+      sortB = sortB.replace(/[$,]/g, '');
+
+      return sortA - sortB;
+    }
+
+    return sortA.localeCompare(sortB);
+  });
+
+  if (prevIndex === titleIndex) {
+    tbody.append(...sorted.reverse());
+    prevIndex = 10;
+  } else {
+    tbody.append(...sorted);
+    prevIndex = titleIndex;
+  }
+});
+
+tbody.addEventListener('dblclick', ev => {
+  const prevContent = ev.target.innerText;
+  let newInput = document.createElement('input');
+
+  newInput.classList.add('cell-input');
+  ev.target.innerText = '';
+  newInput.type = 'text';
+  newInput.value = '';
+
+  if (ev.target.cellIndex === 3) {
+    newInput.type = 'number';
+    newInput.min = 18;
+    newInput.max = 90;
+  }
+
+  if (ev.target.cellIndex === 4) {
+    newInput.type = 'number';
+    newInput.min = 0;
+  }
+
+  if (ev.target.cellIndex === 2) {
+    const select = document.createElement('select');
+
+    select.dataset.qa = 'office';
+
+    select.insertAdjacentHTML('afterbegin', `
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+      `);
+    newInput = select;
+  }
+
+  ev.target.append(newInput);
+  newInput.focus();
+
+  newInput.addEventListener('blur', () => {
+    if (newInput.value === '') {
+      ev.target.innerText = prevContent;
+
+      return;
+    }
+
+    if (newInput.value < 18 || newInput.value > 90) {
+      ev.target.innerText = prevContent;
+      pushNotification('warning', 'Warning', 'Age is incorrect !');
```

Comment:
better to show for user not just 'Age is incorrect !', but what it should be

Reply:
OK. I will add information.